internal/provisioner/utility/unmanaged.go

```diff
@@ -29,14 +56,218 @@ func (u *unmanaged) ValuesPath() string {
 }
 
 func (u *unmanaged) CreateOrUpgrade() error {
-	u.logger.WithField("unmanaged-action", "create").Info("Utility is unmanaged; skippping...")
+	u.logger.WithField("unmanaged-action", "create").Info("Utility is unmanaged; deploying with argocd...")
+
+	privateDomainName, err := u.awsClient.GetPrivateZoneDomainName(u.logger)
+	if err != nil {
+		return errors.Wrap(err, "unable to lookup private zone name")
+	}
+
+	k8sClient, err := k8s.NewFromFile(u.kubeconfigPath, u.logger)
+	if err != nil {
+		return errors.Wrap(err, "failed to set up the k8s client")
+	}
+
+	switch u.Name() {
+	case model.PgbouncerCanonicalName:
+		err = deployManifests(k8sClient, u.logger)
+		if err != nil {
+			return err
+		}
+		if err = u.utiliyArgocdDeploy(u.Name()); err != nil {
+			return errors.Wrapf(err, "failed to provision %s utility", u.Name())
+		}
+
```

Comment:
So we don't really want to overload the `unmanaged` object like this. Ideally, you use the `unmanaged` object directly for utilities that don't have any special creation or deletion logic. If we need that logic like what is going on here then it probably makes more sense to have things like `unmanagedPgbouncer` and `unamangedNginx` just like we have for the normal managed handles.

I know that we want to get this is so I won't make this a blocker, but it should be cleaned up later.

Reply:
Understood, I'll try to change it quickly but we can add a task to make it better.